### PR TITLE
Deploy `front-edge` on kubernetes

### PR DESCRIPTION
**Pull request by fontanierh**
### Initial cluster setup

(assumes the `dust-kube` cluster is already created, the `dust-kube-ingress-ip-address` static IP is already reserved, the DNS A record for `front-edge` is already pointing to that static IP and your local `kubectl` is already connected to the cluster)

from the `k8s` subdirectory, run:
`DD_API_KEY=$DATADOG_API_KEY DD_APP_KEY=$DATADOG_APP_KEY ./apply_infra.sh`

### Setting the secrets

from the `front` subdirectory, assuming you have a valid `.env` file, run:
`./env-to-secret.sh $YOUR_ENV_FILE`

(note: the env file cannot be self referential. If you have an entry such as `URL2=$URL`, the value for `URL2` will not be equal to the value of the `URL` variable, but to `"$URL"`)

### Deploying code

The GitHub action (triggered by manual dispatch) takes care of deploying the latest main onto the cluster.

### Misc

This PR includes:
- Docker files for `connectors` (`web` and `worker`) and `front`
- GCP Cloud build scripts to build `connectors` (`web` and `worker`) and `front`
- kubernetes YAMLs for `front-edge` (deployed as `front-edge.dust.tt`)
- a YAML with our Datadog agent config
- A shell script to sync the kubernetes resources with the YAMLs, and install the datadog agent
- A shell script to deploy `front-edge` on kubernetes
- A GitHub action to deploy `front-edge` on kubernetes
- A shell script to populate `front` secrets from a .env file

### CODE REVIEW

File: front/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM node:18.15.0-alpine as front
```
Comment (fontanierh) [inline front/Dockerfile:1]: it's a very bare-bone Dockerfiles, there might be some ways to improve build speed (I'm not yet a Docker expert)

File: .github/workflows/deploy-front-edge.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        uses: actions/checkout@v2
+
+      - name: Setup Google Cloud CLI
+        uses: google-github-actions/setup-gcloud@v0.2.1
+        with:
+          project_id: ${{ secrets.GCLOUD_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCLOUD_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Setup kubectl
```
Comment (fontanierh) [inline .github/workflows/deploy-front-edge.yml:25]: note: if I am not mistaken, `kubectl` is already pre-installed on GH's `ubuntu-latest`

File: .github/workflows/deploy-front-edge.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        run: |
+          gcloud container clusters get-credentials dust-kube --region us-central1
+
+      - name: Build the image on Cloud Build
```
Comment (fontanierh) [inline .github/workflows/deploy-front-edge.yml:29]: We technically dont _have_ to use cloud build. I already implemented it like this for local use, so I kept it as-is, but it seems a bit slow so I might change it to build directly inside the action

File: k8s/configmaps/front-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: front-config
+data:
+  DD_ENV: "prod"
+  DD_SERVICE: "front-edge"
```
Comment (spolu) [inline k8s/configmaps/front-configmap.yaml:7]: should this be `front-edge-configmap.yaml` in prevision of having both `front` and `front-edge` ?

**Review by spolu — CHANGES_REQUESTED** [review]
Looks great! 

Let's rename all resources that will need to be duplicated for `front` into `front-edge`. Update scripts, PR description and runbooks accordingly :+1:

**Comment by fontanierh** [comment]
@spolu  PR and OP updated. Updating runbook now 👍

**Review by spolu — CHANGES_REQUESTED** [review]
Last `nit` 

move `cloud-build.sh` `cloudbuild.yaml` `deploy-edge.sh` `env-to-secret.sh` to `front/admin` ?

**Review by spolu — CHANGES_REQUESTED** [review]
Last `nit` 

move `cloud-build.sh` `cloudbuild.yaml` `deploy-edge.sh` `env-to-secret.sh` to `front/admin` ?

**Comment by fontanierh** [comment]
moved @spolu

**Review by spolu — APPROVED** [review]
:shipit:

**Pull request merged**